### ISSUE TITLE
[Feature Request] Advanced custom products

### ISSUE THREAD

**Issue by BradBot1** [FOSSBilling/FOSSBilling#1110]
**Is your feature request related to a problem? Please describe.**
Custom products are a great catch all for when there isn't an applicable product however they are super limiting as they are completely manual.

**Describe the solution you'd like**
The ability to either run php code or send custom HTTP requests when a product is purchased, renewed and expired.

**Describe alternatives you've considered**
I have looked into creating a custom addon but it seems extreemly overkill to create one for every possible item I could wish to add.

**Additional context**
Even something as simple as a webhook system would suffice, where data is sent upon a purchase to a pre-defined URL

**Comment by evrifaessa** [comment]
I think it makes the most sense to rename the current "Custom" one to "Manual activation" and create a new one for webhooks. The idea makes sense, sometimes a full adapter is an overkill.

**Comment by BradBot1** [comment]
I have been working on this myself and I have found an _"complication"_.
Products provided inside the dropdown on the admin interface are hard coded:
![image](https://github.com/FOSSBilling/FOSSBilling/assets/45739045/bbd8df77-ab74-4415-b831-9f33f66cf2e3)
https://github.com/FOSSBilling/FOSSBilling/blob/49f00c64aae18ed7df56ed844e811e0667885a58/src/modules/Product/Service.php#L125

Unsure if this should be moved to a different feature request but modularising this would make custom products a lot easier to implement without the whole plugin mess

**Comment by BelleNottelling** [comment]
> Products provided inside the dropdown on the admin interface are hard coded:

Nope. That's not true. The base list is hard-coded (I don't know why, it doesn't really need to be), but it loads additional items if the module id starts with "service". Such as `servicedomain`, `servicehosting`, or `servicecustom`. This is done within the function you linked to.
You can verify this yourself by enabling the "Membership Product" extension: 
![image](https://github.com/FOSSBilling/FOSSBilling/assets/17304943/acec0459-7bc9-43ab-b4a8-efe32826f258)

**Comment by BradBot1** [comment]
I am blind. 🧑‍🦯

**Comment by none0nfg** [comment]
+1, webhook with information about purchased product really usefull
